src/components/implementation/sched/pfprr_quantum_static/main.c

```diff
@@ -630,48 +574,31 @@ void
 slm_ipi_process(void *d)
 {
 	int rcvd, ret;
-	struct slm_ipithd *r = ss_ipithd_get(cos_cpuid()+1);
-	struct slm_notify notify = { 0 };
-	struct slm_thd *current = slm_thd_current();
-	struct slm_thd *thd;
+	struct slm_ipi_percore *ipi_data = slm_ipi_percore_get(cos_cpuid());
+	struct slm_ipi_thd     *r        = &ipi_data->ipi_thd;
+	struct slm_ipi_event    event    = { 0 };
+	struct slm_thd         *current  = slm_thd_current();
+	struct slm_thd         *thd;
 
 	while (1) {
 		cos_rcv(r->rcv, RCV_ALL_PENDING, &rcvd);
-		assert(!slm_notify_queue_empty(&slm_xcpu_notify[cos_cpuid()]));
+		assert(!slm_ipi_event_empty(cos_cpuid()));
 
-		slm_cs_enter(current, SLM_CS_NONE);
-		while (!slm_notify_queue_empty(&slm_xcpu_notify[cos_cpuid()])) {
-			slm_notify_queue_dequeue(&slm_xcpu_notify[cos_cpuid()], &notify);
-			thd = slm_thd_static_cm_lookup(notify.tid);
+		while (!slm_ipi_event_empty(cos_cpuid())) {
+			slm_ipi_event_dequeue(&event, cos_cpuid());
+			thd = slm_thd_static_cm_lookup(event.tid);
+			slm_cs_enter(current, SLM_CS_NONE);
```

Comment:
Lets look at the options, ranked from best to worst. I could be wrong on the ranking, but I figured you might want to see optiosn and why they are good/bad:

1. assert out in the error case. We are assuming that `slm_thd_wakeup` should *not* return an error as we derived `thd` from scheduler-internal data-structures.  This is not strictly correct in all cases as the thread might have been deleted before this call, but then we have larger restructuring issues. Given the sassumptions of the current system, it feels correct to just assert here.
2. Log the error, drop the notification and move on to the next iteration. I don't like this option as having to look at a log to discern a logic error is not good. I think it is better than silently failing.
3. return from this function silently. This is bad as it means developers will see errors that are unrelated to this loop (wakeups stop firing, e.g. "when I take a lock, it never wakes up me...thus the error must be in the lock"), so it will be hard for them to zero-in on this code.